### PR TITLE
[FOGL-1696]- Applied min, max and length checking for config item 

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html [hunk 1]
```diff
@@ -17,31 +17,38 @@
...
             </div>
           </div>
         </ng-container>
-        &nbsp;
         <div class="field">
           <div *ngIf="!f.valid" class="error">
             <sup>*</sup> Field can't left blank.
           </div>
+          <div *ngIf="!validRange" class="error">
+            <sup>*</sup> Field size out of range.
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html]: what is that range? if we are throwing this error  message, we should know `Value should be in range <min>-<max>`

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts [hunk 2]
```diff
@@ -122,4 +127,16 @@ export class ViewConfigItemComponent implements OnInit, OnChanges {
   public getConfigAttributeType(key) {
     return ConfigTypeValidation.getValueType(key);
   }
+
+  public inputValidator(event: any) {
+    const max = event.target.max;
+    const min = event.target.min;
+    const value = event.target.value;
+    if (+value < +min || +value > +max) {
+      console.log('this.configForm.invalid', this.configForm.invalid);
```
Comment (monika-Dianomic) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts]: Please remove this log.

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts [hunk 3]
```diff
@@ -122,4 +127,16 @@ export class ViewConfigItemComponent implements OnInit, OnChanges {
...
+      return this.configForm.invalid;
+    }
+    console.log('this.configForm.valid', this.configForm.valid);
```
Comment (monika-Dianomic) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts]: remove this log.

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html [hunk 4]
```diff
@@ -40,10 +45,12 @@
...
           </div>
         </ng-container>
-        &nbsp;
         <div class="field">
           <div *ngIf="!f.valid" class="error">
-            <sup>*</sup> Field can't left blank.
+            <sup>*</sup> Field can't be left blank.
+          </div>
+          <div *ngIf="!validRange" class="error">
+            <sup>*</sup> Field size out of range.
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html]: -  Value should be in range <min>-<max>
-  Value should not be less than <min>
-  Value should not be greater than <max>

File: src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html [hunk 5]
```diff
@@ -40,10 +45,12 @@
             </div>
           </div>
         </ng-container>
-        &nbsp;
         <div class="field">
           <div *ngIf="!f.valid" class="error">
-            <sup>*</sup> Field can't left blank.
+            <sup>*</sup> Field can't be left blank.
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html]: We need to confirm this, config item allows itself to be blank. 

*And DELETE call reset to its default value

File: src/app/components/core/configuration-manager/configuration-type-validation.ts [hunk 6]
```diff
@@ -41,13 +41,4 @@ export default class ConfigTypeValidation {
     }
     return valueType;
   }
-
-  public static isValidJsonString(str) {
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-type-validation.ts]: are we not using it anymore? or its now a pipe?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 7]
```diff
@@ -1,9 +1,8 @@
 import { Component, Input, OnInit, ViewChild } from '@angular/core';
-import _ from 'lodash-es/array';
+import { TreeComponent } from 'angular-tree-component';
 import { isEmpty } from 'lodash-es/';
-
+import _ from 'lodash-es/array';
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts]: why we have mix usage of lodash and lodash-es?!

File: src/app/components/core/configuration-manager/configuration-type-validation.ts (hunk 6 above)
Comment (praveen-garg) [reply src/app/components/core/configuration-manager/configuration-type-validation.ts]: Okay, its a custom-json-validator.directive now

**Pull request merged**